components/script/devtools.rs

```diff
@@ -250,16 +250,21 @@ pub fn handle_wants_live_notifications(global: &GlobalScope, send_notifications:
 }
 
 pub fn handle_set_timeline_markers(context: &BrowsingContext,
+                                   pipeline: PipelineId,
                                    marker_types: Vec<TimelineMarkerType>,
-                                   reply: IpcSender<TimelineMarker>) {
-    let window = context.active_window();
-    window.set_devtools_timeline_markers(marker_types, reply);
+                                   reply: IpcSender<Option<TimelineMarker>>) {
+    match context.find(pipeline) {
+        None => reply.send(None).unwrap(),
```

Comment:
What use does the receiver have for a `None`? What if we only sent  the timeline marker if we find the pipeline, and therefore still have `IpcSender<TimelineMarker>` instead of introducing an `Option` in there?